Model/Product/Builder.php

```diff
@@ -174,6 +180,25 @@ public function buildImageUrl(Product $product, StoreInterface $store)
     }
 
     /**
+     * Adds the alternative image urls
+     *
+     * @param Product $product the product model.
+     * @return array
+     */
+    protected function buildAlternativeImages(Product $product)
+    {
+        $images = [];
+        $this->galleryReadHandler->execute($product);
+        foreach ($product->getMediaGalleryImages() as $image) {
```

Comment:
Is the product's main image part of the list getMediaGalleryImages?
Should we add a check that it's not?